src/pages/SkillCalculator.js

```diff
@@ -387,3 +389,20 @@ function applyFilters(actions, currentLevel, useLevelFilter, isSkillingProdigy,
     }
     return filteredActions;
 }
+
+function columnsWithData(columns, data) {
+    const hasNoInputs = !data.some((datum) => datum.inputs.length > 0);
+    const hasNoOutputs = !data.some((datum) => datum.outputs.length > 0);
```

Comment:
I wonder if it'd be cleaner to move these checks up to where the columns are defined and set the `hidden` prop on the column (https://react-bootstrap-table.github.io/react-bootstrap-table2/docs/column-props.html#columnhidden-bool )? So then we don't need the whole function to do all the index finding and splicing.

something like
```
 const columns = [
        ...
        {
            "dataField": "inputs",
            "text": "Inputs",
            ....
            "hidden": !data.some((datum) => datum.inputs.length > 0)
        },
        {
            "dataField": "outputs",
            "text": "Outputs",
            ....
            "hidden": !data.some((datum) => datum.outputs.length > 0)
        },
       ....
]
```

Reply:
Nice, that's way easier. Updated  in 391fbb4.